### PR TITLE
CNV#30165: Updated hot plug NIC docs

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s):4.15+

Issue: [CNV-30165](https://issues.redhat.com//browse/CNV-30165) and [CNV-30168](https://issues.redhat.com//browse/CNV-30168)

Link to docs preview: https://71247--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-hot-plugging-network-interfaces

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by phoracek** [comment]
Looks good to me. I'd like @AlonaKaplan to give the final verdict from the dev side

File: modules/virt-hot-unplugging-bridge-network-interface-cli.adoc [hunk 1]
```diff
@@ -16,12 +21,35 @@ You can remove a bridge network interface from a running virtual machine (VM).
...
+            bridge: {}
+    networks:
+      - name: defaultnetwork
+        pod: {}
+      - name: <secondary_nic>
+        multus:
+          networkName: <nad_name>
+# ...
 ----
+<1> Set the interface state to `absent`.
```
Comment (AlonaKaplan) [inline modules/virt-hot-unplugging-bridge-network-interface-cli.adoc:52]: I would also mention that removing the interface from the spec wouldn't cause hot-unplug.
`Set the interface state to `absent` (removing the interface from the spec wouldn't cause hot-unplug)`

Comment (sjhala-ccs) [reply modules/virt-hot-unplugging-bridge-network-interface-cli.adoc:52]: @AlonaKaplan In step 1 of this procedure, I have mentioned "This detaches the network interface from the guest, but the interface still exists in the pod." Do you want me to add that sentence to the callout <1> description as well?

Comment (AlonaKaplan) [reply modules/virt-hot-unplugging-bridge-network-interface-cli.adoc:52]: I just want to emphasize that only marking the interface as absent will detach the interface from a running guest.
Removing the interface completely for the spec won't do it.

**Review by AlonaKaplan — APPROVED** [review]

**Comment by AlonaKaplan** [comment]
Looks good to me, thanks.

**Comment by sjhala-ccs** [comment]
@segevyos Can you please review? Thanks!

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc [hunk 2]
```diff
@@ -22,21 +23,46 @@ Hot plug a bridge network interface to a virtual machine (VM) while the VM is ru
...
 ----
-+
-where:
-
-<vm_name>:: Specifies the name of the `VirtualMachine` object.
-<net_attach_def_name>::  Specifies the name of `NetworkAttachmentDefinition` object.
-<interface_name>:: Specifies the name of the new network interface.
+<1> Specifies the name of the new network interface.
+<2> Specifies the name of the network. This must be the same as the value of the `template.spec.domain.devices.interfaces.name` field. 
+<3> Specifies the name of `NetworkAttachmentDefinition` object.
```
Comment (agantony) [inline modules/virt-hot-plugging-bridge-network-interface-cli.adoc:57]:
```suggestion
This adds the new network interface to the VM and virtual machine instance (VMI) specification but does not attach it to the running VM.
```

File: modules/virt-hot-unplugging-bridge-network-interface-cli.adoc [hunk 3]
```diff
@@ -16,12 +21,35 @@ You can remove a bridge network interface from a running virtual machine (VM).
 
 .Procedure
 
-. Hot unplug a bridge network interface by running the following command. The `virtctl removeinterface` command detaches the network interface from the guest, but the interface still exists in the pod.
+. Hot unplug a secondary network interface by setting the interface state to `absent` in the VM specification. This detaches the network interface from the guest, but the interface still exists in the pod.
```
Comment (agantony) [inline modules/virt-hot-unplugging-bridge-network-interface-cli.adoc:24]:
```suggestion
. To hot unplug a secondary network interface, set the interface state to `absent` in the VM specification:
```

File: modules/virt-hot-unplugging-bridge-network-interface-cli.adoc [hunk 4]
```diff
@@ -16,12 +21,35 @@ You can remove a bridge network interface from a running virtual machine (VM).
...
+            bridge: {}
+    networks:
+      - name: defaultnetwork
+        pod: {}
+      - name: <secondary_nic>
+        multus:
+          networkName: <nad_name>
+# ...
 ----
+<1> Set the interface state to `absent` to detach it from the running VM. Simply removing the interface details from the VM specification does not hot unplug the secondary network interface.
```
Comment (agantony) [inline modules/virt-hot-unplugging-bridge-network-interface-cli.adoc:53]:
```suggestion
This detaches the network interface from the guest, but the interface still exists in the pod.
```

File: modules/virt-hot-unplugging-bridge-network-interface-cli.adoc [hunk 5]
```diff
@@ -16,12 +21,35 @@ You can remove a bridge network interface from a running virtual machine (VM).
...
```
Comment (agantony) [inline modules/virt-hot-unplugging-bridge-network-interface-cli.adoc:52]:
```suggestion
<1> Set the interface state to `absent` to detach it from the running VM. Removing the interface details from the VM specification does not hot unplug the secondary network interface.
```

**Review by agantony — COMMENTED** [review]
Added a few suggestions for your consideration; overall lgtm!

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc [hunk 6]
```diff
@@ -22,21 +23,46 @@ Hot plug a bridge network interface to a virtual machine (VM) while the VM is ru
...
-where:
-
-<vm_name>:: Specifies the name of the `VirtualMachine` object.
-<net_attach_def_name>::  Specifies the name of `NetworkAttachmentDefinition` object.
-<interface_name>:: Specifies the name of the new network interface.
+<1> Specifies the name of the new network interface.
+<2> Specifies the name of the network. This must be the same as the value of the `template.spec.domain.devices.interfaces.name` field. 
+<3> Specifies the name of `NetworkAttachmentDefinition` object. 
 
+. Apply the VM configuration:
```
Comment (segevyos) [inline modules/virt-hot-plugging-bridge-network-interface-cli.adoc:58]: This is wrong.
The VM manifest was already applied, and the VM is running at this point.
The changes (the new interface) should be added via `oc edit vm`. I don't think that the VMI is updated this way.
@AlonaKaplan Please keep me honest here. Thank you.

File: modules/virt-hot-unplugging-bridge-network-interface-cli.adoc [hunk 7]
```diff
@@ -16,12 +21,35 @@ You can remove a bridge network interface from a running virtual machine (VM).
 
 .Procedure
 
-. Hot unplug a bridge network interface by running the following command. The `virtctl removeinterface` command detaches the network interface from the guest, but the interface still exists in the pod.
+. Hot unplug a secondary network interface by setting the interface state to `absent` in the VM specification. This detaches the network interface from the guest, but the interface still exists in the pod.
 +
```
Comment (segevyos) [inline modules/virt-hot-unplugging-bridge-network-interface-cli.adoc:25]: Again - `oc edit vm` is required.

**Review by segevyos — CHANGES_REQUESTED** [review]

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc [hunk 8]
```diff
@@ -22,21 +23,46 @@ Hot plug a bridge network interface to a virtual machine (VM) while the VM is ru
...
+metadata:
+  name: vm-fedora
+template:
+  spec:
+    domain:
+      devices:
+        interfaces:
+        - name: defaultnetwork
+          masquerade: {}
+          # new interface
```
[the remark this answers is not in the document]
Comment (sjhala-ccs) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc:42]: I have adjusted the indentation for both comments

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc (hunk 6 above)
Comment (sjhala-ccs) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc:58]: @AlonaKaplan Can you please respond?

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc [hunk 9]
```diff
@@ -22,21 +23,46 @@ Hot plug a bridge network interface to a virtual machine (VM) while the VM is ru
 $ virtctl start <vm_name>
 ----
 
-. Use the following command to hot plug a new network interface to the running VM. The `virtctl addinterface` command adds the new network interface to the VM and virtual machine instance (VMI) specification but does not attach it to the running VM.
+. Add the new network interface to the running VM by editing the VM specification. This adds the new network interface to the VM and virtual machine instance (VMI) specification but does not attach it to the running VM.
```
[the remark this answers is not in the document]
Comment (sjhala-ccs) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc:26]: Thanks for the suggestion, but I would like to keep the sentence structure as is for both modules. IMO, the second sentence needs to be in its current position for clarity of context.

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc [hunk 10]
```diff
@@ -22,21 +23,46 @@ Hot plug a bridge network interface to a virtual machine (VM) while the VM is ru
...
+<1> Specifies the name of the new network interface.
+<2> Specifies the name of the network. This must be the same as the `name` of the new network interface that you defined in the `template.spec.domain.devices.interfaces` list. 
+<3> Specifies the name of the `NetworkAttachmentDefinition` object. 
 
+. Apply the VM configuration by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f vm-fedora.yaml
+----
```
Comment (ormergi) [inline modules/virt-hot-plugging-bridge-network-interface-cli.adoc:63]: I guess this should be removed, now that the doc suggest using `oc edit`

Comment (ormergi) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc:63]: I got it wrong, it doesn't specify using 'oc edit'

Comment (segevyos) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc:63]: > I got it wrong, it doesn't specify using 'oc edit'

But it should

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc (hunk 6 above)
Comment (AlonaKaplan) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc]: `oc apply -f` will work, but indeed it is better to use `oc edit`.

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc (hunk 10 above)
Comment (ormergi) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc:63]: I guess the reason is the fact we dont know weather VM created with `create` or `apply` command, right?

File: modules/virt-hot-plugging-bridge-network-interface-cli.adoc (hunk 6 above)
Comment (segevyos) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc]: It didn't work for me, even though I created the VM with `oc apply -f`.

Comment (ormergi) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc]: Could you please share the error you got?

Comment (sjhala-ccs) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc]: I have updated the docs to use `oc edit`

Comment (segevyos) [reply modules/virt-hot-plugging-bridge-network-interface-cli.adoc]: > Could you please share the error you got?

I didn't see any error, just that the VMI spec was not updated, thus the guest VM was not updated as well (after migration).

**Pull request merged**